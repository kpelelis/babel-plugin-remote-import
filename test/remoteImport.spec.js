import pluginTester from "babel-plugin-tester";
import prettier from "prettier";
import plugin from "../src";

jest.mock("https");

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename
  },
  formatResult(result) {
    return prettier.format(result, { trailingComma: "es5" });
  },
  tests: {
    "Correct usage": {
      error: false,
      code: `
        import image from 'https://i.imgur.com/hQVq8qN.jpg'
        
        console.log(image)
      `
    },
    "Should not replace path": {
      error: false,
      code: `
        import image from './test.js'
        
        console.log(image)
      `
    },
    "Incorrect path": {
      error: false,
      code: `
        import image from 'https://www.google.com'
        
        console.log(image)
      `
    }
  }
});
