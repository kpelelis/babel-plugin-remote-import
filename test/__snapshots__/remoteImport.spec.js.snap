// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remote-import Correct usage: Correct usage 1`] = `
"
import image from 'https://i.imgur.com/hQVq8qN.jpg'

console.log(image)

      ↓ ↓ ↓ ↓ ↓ ↓

import image from \\"/var/folders/rh/w4hpfkkn4q54mr1hmldsxj7h0000gp/T/hQVq8qN.jpg\\";

console.log(image);

"
`;

exports[`remote-import Incorrect path: Incorrect path 1`] = `
"
import image from 'https://www.google.com'

console.log(image)

      ↓ ↓ ↓ ↓ ↓ ↓

import image from \\"https://www.google.com\\";

console.log(image);

"
`;

exports[`remote-import Should not replace path: Should not replace path 1`] = `
"
import image from './test.js'

console.log(image)

      ↓ ↓ ↓ ↓ ↓ ↓

import image from \\"./test.js\\";

console.log(image);

"
`;
